scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/util/DataValidationHelper.java

```diff
@@ -55,32 +56,37 @@ public static boolean isValidTimeZone(String timeZoneId) {
 	}
 
 	/**
-	 * Checks if the specified time in milli seconds is after now (current time).
-	 * 
-	 * @param timeInMillis
+	 * Checks if the specified date time is after now (current time).
+	 * @param dateTime
 	 * @param timeZone
-	 * 
 	 * @return
 	 */
-	public static boolean isLaterThanNow(Long timeInMillis, TimeZone timeZone) {
-		if (timeInMillis != null && timeZone != null) {
-			Calendar calendar = Calendar.getInstance(timeZone);
-			Long nowInMillis = calendar.getTimeInMillis();
-
-			return timeInMillis > nowInMillis;
+	public static boolean isLaterThanNow(Date dateTime, TimeZone timeZone) {
```

Comment:
Thanks for the heads up Michael.
Yes you are right. In the process of refactoring missed these points , will fix them in next refactoring.

Regards,

Harpreet Ghai
[cid:image001.jpg@01D1EE23.7E9A9120]
[cid:image002.jpg@01D1EE23.7E9A9120]
From: Michael Fraenkel [mailto:notifications@github.com]
Sent: Thursday, 4 August 2016 1:30 AM
To: cloudfoundry-incubator/app-autoscaler
Cc: Ghai, Harpreet; Author
Subject: Re: [cloudfoundry-incubator/app-autoscaler] [#127046799] - Specific Date schedule should have the start date and time merged into a single property, also the end date and time (#63)


In scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/util/DataValidationHelper.java<https://github.com/cloudfoundry-incubator/app-autoscaler/pull/63#discussion_r73350473>:

>        * @return

>        */

> -     public static boolean isLaterThanNow(Long timeInMillis, TimeZone timeZone) {

> -            if (timeInMillis != null && timeZone != null) {

> -                    Calendar calendar = Calendar.getInstance(timeZone);

> -                    Long nowInMillis = calendar.getTimeInMillis();

> -

> -                    return timeInMillis > nowInMillis;

> +     public static boolean isLaterThanNow(Date dateTime, TimeZone timeZone) {

Just a general comment for future reference.

Your naming is a bit mixed. Here you use isLaterThanNow and below you use isAfter. Shouldn't this be isAfterNow?

Why don't you just do: return calToCompare.after(calNow); instead of the if block?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/cloudfoundry-incubator/app-autoscaler/pull/63/files/9cce4e237acece97c87c31de157c9511d6451293#r73350473>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AMmaH8TikYrUuN75f8N4TheUJWzKeLscks5qcLP_gaJpZM4JaQWr>.

Disclaimer

The information in this e-mail is confidential and may contain content that is subject to copyright and/or is commercial-in-confidence and is intended only for the use of the above named addressee. If you are not the intended recipient, you are hereby notified that dissemination, copying or use of the information is strictly prohibited. If you have received this e-mail in error, please telephone Fujitsu Australia Software Technology Pty Ltd on + 61 2 9452 9000 or by reply e-mail to the sender and delete the document and all copies thereof.


Whereas Fujitsu Australia Software Technology Pty Ltd would not knowingly transmit a virus within an email communication, it is the receiver’s responsibility to scan all communication and any files attached for computer viruses and other defects. Fujitsu Australia Software Technology Pty Ltd does not accept liability for any loss or damage (whether direct, indirect, consequential or economic) however caused, and whether by negligence or otherwise, which may result directly or indirectly from this communication or any files attached.


If you do not wish to receive commercial and/or marketing email messages from Fujitsu Australia Software Technology Pty Ltd, please email unsubscribe@fast.au.fujitsu.com